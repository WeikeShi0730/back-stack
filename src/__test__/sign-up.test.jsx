import { render, screen } from "@testing-library/react";
import SignUp from "../components/sign-up/sign-up.component";

let documentBody;
describe("<SignUp />", () => {
  it("Test <SignUp />", () => {
    documentBody = render(<SignUp />);
    expect(
      screen.getByRole("button", { name: /Sign up/i })
    ).toBeInTheDocument();

    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
