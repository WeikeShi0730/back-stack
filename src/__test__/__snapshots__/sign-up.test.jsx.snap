// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> Test <SignUp /> 1`] = `
<body>
  <div>
    <div
      class="max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto mt-10 mb-4 bg-white rounded-lg border border-primaryBorder shadow-default py-8 px-10"
    >
      <h1
        class="text-base lg:text-lg font-light text-primary mt-4 mb-12 text-center"
      >
        Don't have an account? Sign up üîê
      </h1>
      <form>
        <div>
          <label
            class="text-sm md:text-base"
          >
            Username
          </label>
          <input
            class="text-xs md:text-md w-full p-2 text-primary border rounded-md outline-none transition duration-150 ease-in-out mb-4"
            maxlength="10"
            name="displayName"
            placeholder="Your username"
            required=""
            type="text"
          />
        </div>
        <div>
          <label
            class="text-sm md:text-base"
          >
            Email
          </label>
          <input
            class="text-xs md:text-md w-full p-2 text-primary border rounded-md outline-none transition duration-150 ease-in-out mb-4"
            name="email"
            placeholder="Your email"
            required=""
            type="email"
          />
        </div>
        <div>
          <label
            class="text-sm md:text-base"
          >
            Password
          </label>
          <input
            autocomplete="off"
            class="text-xs md:text-md w-full p-2 text-primary border rounded-md outline-none transition duration-150 ease-in-out mb-4"
            minlength="6"
            name="password"
            placeholder="Your password"
            required=""
            type="password"
          />
        </div>
        <div
          class="flex justify-center items-center mt-6"
        >
          <button
            class="text-xs md:text-sm bg-gray-800 py-2 px-4 text-white rounded border focus:outline-none font-light"
          >
            Sign up
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`;
